src/planwise/component/importer.clj

```diff
@@ -190,19 +191,38 @@
       (reduce reducer-fn {} jobs))))
 
 ;; ----------------------------------------------------------------------------
+;; Jobs persistance
+
+(defn- persist-job
+  [component job]
+  (let [dataset-id (import-job/job-dataset-id job)]
+    (datasets/update-dataset (:datasets component)
+                             {:id dataset-id
+                              :import-job (dissoc job :fsm)})))
```

Comment:
I'd encapsulate the job internal structure in the `import-job` namespace and provide a `serialize-job` or similar. That way it's also symmetric with `restore-job` below.